src/Common/Telemetry/Migration.php

```diff
@@ -0,0 +1,320 @@
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	public $fs_plugins_slug = 'tec_freemius_plugins_archive';
+
+	/**
+	 * Placeholder for fs_accounts data
+	 *
+	 * @since TBD
+	 *
+	 * @var [type]
+	 */
+	private $fs_accounts;
+
+	/**
+	 * List of our plugins to check for.
+	 *
+	 * @since TBD
+	 *
+	 * @var array
+	 */
+	public $our_plugins = [
+		'the-events-calendar',
+		'event-tickets'
+	];
+
+	/**
+	 * Determine if we are opted-in to Freemius
+	 *
+	 * @since 5.0.17
+	 *
+	 * @return boolean
+	 */
+	public function is_opted_in(): bool {
+		global $wpdb;
+		$fs_accounts = $wpdb->get_results( "SELECT `option_value` FROM $wpdb->options WHERE `option_name` = 'fs_accounts'", ARRAY_A );
+
+		if ( empty( $fs_accounts ) || $fs_accounts instanceof \WP_Error ) {
+			return false;
+		}
+
+		$fs_accounts = array_pop( $fs_accounts );
+
+		// Prevent issues with incomplete classes
+		$result = preg_replace(
+			'/O:(\d+):"(?:[^:]+)":/m',
+			'a:',
+			$fs_accounts['option_value']
+		);
+		$result = substr( $result, stripos( $result, 'sites' ) );
+
+		foreach ( $this->our_plugins as $plugin ) {
+			$pos = stripos( $result, $plugin );
+			if ( false !== $pos ) {
+				$pos = stripos( $result, 'is_disconnected', $pos );
+				if ( false !== $pos ) {
+					$pos = stripos( $result, 'b:', $pos );
+					if ( false !== $pos ) {
+						$pos = stripos( $result, '0', $pos );
+						if ( false !== $pos ) {
+							return true;
+						}
+					}
+				}
+			}
+		}
+
+		return false;
+	}
+
+	/**
+	 * Whether the class should load/run.
+	 *
+	 * @since 5.0.17
+	 *
+	 * @return boolean
+	 */
+	public function should_load(): bool {
+		// If we're not opted in to Freemius, bail.
+		if ( ! $this->is_opted_in() ) {
+			return false;
+		}
+
+		/**
+		 * Allows filtering of whether the class should load/run.
+		 */
+		return apply_filters( 'tec_telemetry_migration_should_load', true );
+	}
+
+	/**
+	 * Detect if the user has opted in to Freemius and auto-opt them in to Telemetry.
+	 *
+	 * @since 5.0.17
+	 */
+	public function migrate_existing_opt_in(): void {
+		// Let's reduce the amount this triggers.
+		if ( defined( 'DOING_AJAX' ) && DOING_AJAX ) {
+			return;
+		}
+
+		if ( defined( 'DOING_AUTOSAVE' ) && DOING_AUTOSAVE ) {
+			return;
+		}
+
+		if ( ! $this->should_load() ) {
+			return;
+		}
+
+		$fs_active_plugins = get_option( 'fs_active_plugins' );
+
+		// Bail if empty.
+		if ( empty( $fs_active_plugins ) ) {
+			return;
+		}
+
+		// Clean up our list.
+		$this->remove_inactive_plugins( $fs_active_plugins );
+
+		// Bail if none of our plugins are present.
+		if ( ! count( $this->our_plugins ) ) {
+			return;
+		}
+
+		/**
+		 * Allows plugins to hook in and perform actions (like display a notice) when
+		 * the user is automatically opted in to Telemetry.
+		 *
+		 * We also use this to trigger the actual auto-opt-in at the default priority.
+		 *
+		 * @since 5.0.17
+		 */
+		do_action( 'tec_telemetry_auto_opt_in' );
+
+		$fs_accounts = get_option( 'fs_accounts' );
```

Comment:
I dont think we can call `get_option()` on that particular Freemius option when the Freemius Codebase is not present, it will try to `unserialize()` when fetching get option which will throw an error because we dont have `FS_Plugin` as a class on our codebase anymore.

We should chat on how to solve this I have a couple options that I was investigating for us while doing the code review.